### PR TITLE
[L2B-6491] Add time-based indexer for Starkex

### PR DESCRIPTION
**Pull request by maciekop-l2b**
Resolves L2B-6491

### CODE REVIEW

File: packages/backend/src/modules/activity2/Activity2Module.ts [hunk 1]
```diff
@@ -36,9 +39,8 @@ export function createActivity2Module(
     logger = logger.for('Activity2Module')
     logger.info('Starting')
```
Comment (antooni) [inline packages/backend/src/modules/activity2/Activity2Module.ts:40]: are those logs needed?

**Review by maciekzygmunt — APPROVED** [review]

File: packages/backend/src/modules/activity2/Activity2Module.ts (hunk 1 above)
Comment (maciekop-l2b) [reply packages/backend/src/modules/activity2/Activity2Module.ts:40]: yes, that's the pattern we're following for all modules

File: packages/backend/src/modules/activity2/indexers/DayTargetIndexer.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { Logger } from '@l2beat/backend-tools'
+import { RootIndexer } from '@l2beat/uif'
+import { Clock } from '../../../tools/Clock'
+
+export class DayTargetIndexer extends RootIndexer {
+  constructor(
+    logger: Logger,
+    private readonly clock: Clock,
+    private readonly module?: string,
```
Comment (antooni) [inline packages/backend/src/modules/activity2/indexers/DayTargetIndexer.ts:9]: not needed

Comment (maciekop-l2b) [reply packages/backend/src/modules/activity2/indexers/DayTargetIndexer.ts:9]: Removed

File: packages/backend/src/modules/activity2/indexers/DayActivityIndexer.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import { DayActivityIndexerDeps } from './types'
+
+export class DayActivityIndexer extends ManagedChildIndexer {
+  constructor(private readonly $: DayActivityIndexerDeps) {
+    const logger = $.logger.tag($.projectId)
+    super({ ...$, logger, name: `activity_indexer_${$.projectId}` })
+  }
+
+  override async update(from: number, to: number): Promise<number> {
+    // TODO: confirm if this logic is still needed
```
Comment (antooni) [inline packages/backend/src/modules/activity2/indexers/DayActivityIndexer.ts:11]: have you confirmed?

File: packages/backend/src/modules/activity2/indexers/DayActivityIndexer.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    // starkex APIs are not stable and can change from the past. With this we make sure to scrape them again
+    const fromWithUncertainty = from - (this.$.uncertaintyBuffer ?? 0)
```
Comment (antooni) [inline packages/backend/src/modules/activity2/indexers/DayActivityIndexer.ts:13]: I would make this `uncertaintyBuffer` required and move this ?? to initialization

File: packages/backend/src/modules/activity2/indexers/DayActivityIndexer.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+      adjustedTo,
+    )
+
+    await this.$.activityRepository.addOrUpdateMany(counts)
+
+    return adjustedTo
+  }
+
+  override invalidate(targetHeight: number): Promise<number> {
+    // no need to delete data as it will be overwritten by new values
```
Comment (antooni) [inline packages/backend/src/modules/activity2/indexers/DayActivityIndexer.ts:33]: with this assumption during resync we will show invalidated values, is it intended?

File: packages/backend/src/modules/activity2/indexers/DayActivityIndexer.ts (hunk 3 above)
Comment (maciekop-l2b) [reply packages/backend/src/modules/activity2/indexers/DayActivityIndexer.ts:11]: We need more time/experiments to confirm this

File: packages/backend/src/modules/activity2/Activity2Module.ts [hunk 6]
```diff
@@ -56,56 +58,90 @@ function createActivityIndexers(
...
+        indexers.push(activityIndexer)
+        break
+      }
+    }
+  })
+
+  return indexers
+}
+
+function getBatchSizeFromCallsPerMinute(
```
Comment (antooni) [inline packages/backend/src/modules/activity2/Activity2Module.ts:138]: move this logic to provider

File: packages/backend/src/modules/activity2/indexers/DayActivityIndexer.ts (hunk 4 above)
Comment (maciekop-l2b) [reply packages/backend/src/modules/activity2/indexers/DayActivityIndexer.ts:13]: Done

File: packages/backend/src/modules/activity2/indexers/DayActivityIndexer.ts (hunk 5 above)
Comment (maciekop-l2b) [reply packages/backend/src/modules/activity2/indexers/DayActivityIndexer.ts:33]: Yes, that is intended

File: packages/backend/src/modules/activity2/Activity2Module.ts (hunk 6 above)
Comment (maciekop-l2b) [reply packages/backend/src/modules/activity2/Activity2Module.ts]: Done

**Review by antooni — APPROVED** [review]

**Pull request merged**